www/js/coolhole-util.js

```diff
@@ -73,6 +93,334 @@ function isMessageTooOld(msgTime) {
+
+    // Create the utterance object for the message with modified rate/pitch/voice/message
+    const utterThis = new SpeechSynthesisUtterance(speechObj.message);
+
+    if (speechObj.voiceObj) {
+      utterThis.voice = speechObj.voiceObj;
+      utterThis.rate = speechObj.rate;
+      utterThis.pitch = speechObj.pitch;
+    }
+
+    // If there any sounds in the array...
+    if (sounds && sounds.length > 0) {
+      // Wait for all sounds to end; prepare for over-engineering
+      // Since we don't know when they start, we can't rely on the duration
+
+      // Setup a count to prevent a runaway interval
+      let intervalCount = 0;
+      // Interval function that either bails if all sounds have ended or we've called it too many times
+      function checkIfAllSoundHaveEnded(sounds) {
+        // We hit our threshold; stop waiting
+        if (intervalCount > 50) {
+          console.error(
+            "formatSecretaryMessage: sound interval never ended; hit interval count threshold",
+            sounds
+          );
+          window.speechSynthesis.speak(utterThis);
+          clearInterval(interval);
+        }
+        // Some sound is still going; keep waiting
+        if (sounds.some((sound) => !sound.ended)) {
+          intervalCount++;
+          return;
+        }
+        // All sounds have ended
+        console.log("All sounds ended; let's bail");
+        window.speechSynthesis.speak(utterThis);
+        clearInterval(interval);
+      }
+
+      // Begin interval with a check every 100ms
+      const interval = setInterval(function () {
+        checkIfAllSoundHaveEnded(sounds);
+      }, 100);
+    }
+    // If there are no sounds, ignore all the complicated shit above
+    else {
+      window.speechSynthesis.speak(utterThis);
+    }
+  }
+  // Delete when animation is finished
+  setTimeout(() => div.remove(), 7000);
+}
+
+/* 
+This function reads the chatMessage and modifies the speechObj based on flags in the chatMessage.
+It expects the flags to be between curly brackets {}.
+-r = rate of the speech. The units expected are decimal/integer. If it can't be parsed, default is 1.2.
+-p = pitch of the speech. The units expected are decimal/integer. If it can't be parsed, decault is 1.
+-v = voice. A string is expected. If it can't be found, Microsoft Zira - English (United States) is the default.
+
+It also strips out the {}'s out of the chatMessage and puts the results in speechObj.message.
+
+Example:
+	chatMessage:
+	/secretary {-r 1.2 -p 3.4 -v "Microsoft Zira - English (United States)"} this is my message.
+
+	speechObj.rate = 1.2
+	speechObj.pitch = 3.4
+	speechObj.voiceObj = (voice object from voices array)
+	speechObj.message = "this is my message."
+*/
+function processSpeechMessage(chatMessage) {
+  const defaultSpeechObj = {
+    rate: 1.2,
+    pitch: 1,
+    voiceObj: voices.find(
+      (voice) => voice.name === "Microsoft Zira - English (United States)"
+    ),
```

Comment:
Not everyone loves CBT/Wangblows
```suggestion
    voiceObj: voiceObj: voices.find((voice) => voice.default) ?? voices[0],
```

Reply:
its good